tests/integration/local/test_shared_lib_include.py

```diff
@@ -0,0 +1,136 @@
+
+        library_path = "tests/integration/local/test_shared_libraries/liba.so"
+
+        library_code = (
+        	"`root-config --cxx` "
+        	"`root-config --cflags --libs` "
+        	"-fPIC -shared {cpp}"
+        	" -o {lib}"
+        ).format(cpp = cpp_path, lib = library_path)
+        # This creates the shared library
+        subprocess.call(library_code, shell = True)
+
+        # Path to the shared library relative to the main PyRDF directory.
+        so_path = (
+        	"tests/integration/local/"
+        	"test_shared_libraries/liba.so"
+        )
+
+        PyRDF.include_shared_libraries(so_path)
+
+        # The user can include directly the header related to the library
+        # or choose to declare functions or objects later
+        header_path = (
+        	"tests/integration/local/"
+        	"test_shared_libraries/a.h"
+        )
+        PyRDF.include_headers(header_path)
+
+        # Creates an RDataFrame with 5 integers [0...4]
+        rdf = PyRDF.RDataFrame(5)
+
+        # This filters out all numbers less than 3
+        filter1 = rdf.Filter("tdfentry_ < 3")
+
+        # This defines a new variable x with all entries squared
+        # then filters out all x values less than 3
+        filter2 = rdf.Define("x","f(tdfentry_)").Filter("x < 3")
+
+        count1 = filter1.Count().GetValue()
+        count2 = filter2.Count().GetValue()
+
+        # The final answer should be the number of integers
+        # less than 5, which is 3, and the number of squared integers less
+        # than 5, which is 2.
+        self.assertEqual(count1, 3)
+        self.assertEqual(count2, 2)
+
+        # Remove library at the end so it doesn't get uploaded with the project
+        os.remove(so_path)
+
+    def test_includes_shared_lib_with_pcm(self):
+        """
+        Check that a pcm file is correctly included with its shared library.
+        """
+        # Get path of current working directory
+        root_dir = os.getcwd()
+        # Change working directory to create shared library
+        wd_path = "tests/integration/local/test_shared_libraries/"
+        os.chdir(wd_path)
+
+        h_path = "myclass.h"
+        cxx_path = "myclass.cxx"
+        library_path = "libmyclass.so"
+        pcm_path = "myclass_rdict.pcm"
+
+        # Create the pcm file from the header
+        pcm_code = "genreflex {h} -o {cxx}".format(h = h_path, cxx = cxx_path)
+        subprocess.call(pcm_code, shell = True)
+
+        # Create the shared library in the same folder as the pcm
+        library_code = (
+            "g++ -shared -fPIC "
+            "`root-config --cflags --libs` "
+            "{cxx} "
+            "-o {lib}"
+        ).format(cxx = cxx_path, lib = library_path)
+        subprocess.call(library_code, shell = True)
+
+        # Change back to initial working directory
+        os.chdir(root_dir)
+
+        # Disable ROOT autoparsing, this way if the pcm is not present it will
+        # trigger an error
```

Comment:
"Disable autoparsing otherwise ROOT, even w/o pcm, will be able through the AST to infer the "Real Data" of the class, i.e. its data member.